include/eve/module/core/regular/impl/is_less.hpp

```diff
@@ -0,0 +1,50 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+namespace eve::detail
+{
+  template<typename T, typename U, callable_options O>
+  EVE_FORCEINLINE constexpr common_logical_t<T, U> is_less_(EVE_REQUIRES(cpu_), O const& o, T a, U b) noexcept
+  {
+    if constexpr (product_type<T> && product_type<U>)
+    {
+      return convert(kumi::to_tuple(a) < kumi::to_tuple(b), as_element<common_logical_t<T, U>>());
```

Comment:
I bet this gives us an extra conversion some case. We should think about this more